browser/main/dataReducer.js

```
@@ -0,0 +1,425 @@
+
+      // Remove folders from folderMap
+      if (storage != null) {
+        state.folderMap = new Map(state.folderMap)
+        storage.folders.forEach(folder => {
+          const folderKey = storage.key + '-' + folder.key
+          state.folderMap.delete(folderKey)
+        })
+      }
+
+      // Remove notes from noteMap and tagNoteMap
+      const storageNoteSet = state.storageNoteMap.get(action.storageKey)
```

Biome (2.5.0)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)